src/core/Rule.ml

```diff
@@ -275,28 +275,13 @@ type extract_rule = extract_mode rule_info [@@deriving show]
 
 (* the general type *)
 type rule = mode rule_info [@@deriving show]
+type mode_type = Match | Extract [@@deriving show]
```

Comment:
(I guess because now we parse rules lazily, but we need to differentiate early whether a rule is
extract or match/taint because for extract rules we need to evaluate them early.)

Maybe change Match to SearchOrTaint